packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -412,6 +413,15 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 				// Only configure this environment if it is a Worker using Node.js compatibility.
 				if (isNodeCompat(getWorkerConfig(name))) {
 					return {
+						build: {
+							rollupOptions: {
+								treeshake: {
+									// We need to ensure that all the Node.js polyfill files that are simply imported with no exports
+									// are considered to have side-effects to avoid them being tree-shaken.
+									moduleSideEffects: nodeJSCompatHasSideEffect,
```

Comment:
I wonder if this would still be necessary if the unenv preset had `sideEffects: true` in its `package.json`?

If not, I think it would be better to use an array of modules here rather than a function so that it can be merged with user config.

```suggestion
									moduleSideEffects: [...nodeCompatPolyfills],
```

Reply:
I forgot about side effects in package.json!
If that doesn't work I'll go with an array. The function was only for debugging really

Reply:
`sideEffects` fixed it!